kubernetes-discovery-client/src/main/java/io/micronaut/kubernetes/client/v1/configmaps/ConfigMapOrStatus.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2017-2021 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.kubernetes.client.v1.configmaps;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import io.micronaut.kubernetes.client.v1.KubernetesObject;
+
+/**
+ * A polymorphic {@link KubernetesObject} representing either a {@link ConfigMap} or a {@link Status}.
+ * When <em>kind</em> is omitted (s.a. when returned as a {@link ConfigMapList} item), it is assumed to be a {@link ConfigMap}.
+ *
+ * @author Régis Desgroppes
+ * @since 2.3.4
+ */
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "kind", defaultImpl = ConfigMap.class)
+@JsonSubTypes({
+        @JsonSubTypes.Type(value = ConfigMap.class, name = "ConfigMap"),
+        @JsonSubTypes.Type(value = Status.class, name = "Status"),
+})
+public abstract class ConfigMapOrStatus extends KubernetesObject {
```

Comment:
This intermediate class is probably too global, which leads to handle the special case where `ConfigMap` has no `kind` when returned as a `ConfigMapList` item. I could simply move `@Json` annotations to the field instead.